tests/index.js

```diff
@@ -732,6 +732,36 @@ describe('Thorn', () => {
 
                 expect(server.isDone()).to.be.true;
             });
+
+            it('should use the internal username for server interactions', function*() {
```

Comment:
On another note, this test file is getting very big. It might be worth considering splitting this up in the near future, but not as part of this PR.